### DISCUSSION TITLE
user population status api with pagination

### DISCUSSION THREAD

**Discussion by elofland** [processone/ejabberd#4042] [Ideas]
I'm looking for a way to get the connected status on a large group of users (potentially millions) without taxing the server(s). The connected_users API only shows the online users, I'm looking to get both the online and offline users and be able to either be able to get only x number of users (assuming a sorted return list - with start / end or start / page size) or be able to query for x number of users by username (supply a list to check for status of each).

**Comment by badlop** [comment]
You can either write a tool to aggregate information from different ejabberd commands ([registered_users](https://docs.ejabberd.im/developer/ejabberd-api/admin-api/#registered-users), [connected_users_info](https://docs.ejabberd.im/developer/ejabberd-api/admin-api/#connected-users-info)), and probably you will still have to write some command to get some information that no existing commands provides...

Or write your own module (in Erlang or Elixir) with your desired API command, reusing code from the existing commands (probably connected_users_info is a good start).
